### PR TITLE
feat: added venue data to confirmation emails

### LINKED ISSUE #2229: Venue data not available to attendees
### Discuss your ideas or share your views: 
I remember some discussion about it, but I'm unable to find it right now.

Venue data (physical venue, streaming url) is not really available to attendees. Outside of the (admin) dashboard, the only moment when attendee is informed about venue data is when it's changed.

It is not included in RSVP confirmation email, or on event page available to attendees. It is however in invite email send to chapter members.

If invite email contains it, then I think there's no reason to not include it in the confirmation email. On the event page, venue data could be displayed only after RSVPing.

### PR DESCRIPTION
**Pull request by ttran913**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

Closes #2229

Venue data will now be visible in attendee's confirmation emails.
Added venue data(physicalLocation, streamingData, start_at, ends_at) to email-templates.ts.
Modified confirmation related functions in event-email.ts to use the mentioned variables.
Modified eventConfirmAttendeeEmail in resolver.ts to grab venue data.
Modfied eventConfirmAttendeeEmail function to properly take in newly created parameters(streaming_url, venue_physical, venue_type, start_at, ends_at) in event-email.tests.ts. The test fails without these parameters.

### CODE REVIEW

File: server/src/util/event-email.ts [hunk 1]
```diff
@@ -263,6 +290,17 @@ export const eventAttendanceConfirmation = ({
...
+
+  // Following the same format from eventInviteEmail,
+  // grabbing physical location and streaming data
+  // in the same manner
+  const physicalLocation = isPhysical(event.venue_type)
+    ? `\n${physicalLocationShortText(event.venue?.name)}`
+    : '';
+  const streamingData = isOnline(event.venue_type)
+    ? `\n${streamingURLText(event.streaming_url)}`
+    : '';
```
Comment (gikf) [inline server/src/util/event-email.ts:302]: These are used in few places, perhaps it's time to add helper function for each of them.

**Review by gikf — COMMENTED** [review]
It's looking well.

I have couple nitpicks. There appear to be few leftover comments in the code.

For the rest see specific comments below.

File: server/src/util/event-email.ts [hunk 2]
```diff
@@ -227,8 +228,32 @@ export const eventCancelationEmail = (event: CancelEvent) => {
...
+// Revamping to include venue data... manually, maybe make an
+// interface later.
+export const eventConfirmAttendeeEmail = (
+  eventName: string,
+  streaming_url: string | null,
+  venue: string | undefined,
+  venue_type: events_venue_type_enum,
+  start_at: Date,
+  ends_at: Date,
+) => {
```
[the remark this answers is not in the document]
Comment (ttran913) [reply server/src/util/event-email.ts]: I have done this with success. "eventConfirmAttendeeEmail" only takes one parameter of "updatedUser.event"

File: server/src/util/event-email.ts (hunk 1 above)
Comment (ttran913) [reply server/src/util/event-email.ts]: I made some helper functions called getPhysicalLocation and getStreamingData near the top. I hope this is what you meant?

**Comment by ttran913** [comment]
We have also removed all of our comments that that we have made.

File: server/src/util/event-email.ts [hunk 3]
```diff
@@ -69,6 +79,16 @@ export const hasStreamingUrlChanged = (
...
 
+export const getStreamingData = (
+  venue_type: events_venue_type_enum,
+  streaming_url: string | null,
+) => {
+  const streamingData = isOnline(venue_type)
+    ? `\n${streamingURLText(streaming_url)}`
+    : '';
+  return streamingData;
+};
```
Comment (gikf) [inline server/src/util/event-email.ts:90]:
```suggestion
) => isOnline(venue_type)
    ? `\n${streamingURLText(streaming_url)}`
    : '';
```

File: server/src/util/event-email.ts [hunk 4]
```diff
@@ -227,8 +246,32 @@ export const eventCancelationEmail = (event: CancelEvent) => {
   );
 };
 
-export const eventConfirmAttendeeEmail = (eventName: string) =>
-  withUnsubscribe(eventConfirmAtendeeText({ eventName }));
+interface ConfirmAttendeeData {
+  name: string;
+  streaming_url: string | null;
+  venue: venues | null;
```
Comment (gikf) [inline server/src/util/event-email.ts:252]:
```suggestion
  venue: { name: string | undefined } | null;
```
`venues` type has few fields that won't be used here at all. `name` is at most what interests us at this point.

**Review by gikf — COMMENTED** [review]
Thanks for the update, it's looking well.

Just a couple nitpick suggestions more.

File: server/src/util/event-email.ts (hunk 4 above)
Comment (ttran913) [reply server/src/util/event-email.ts]: Thank you for that! I was having trouble in resolver.ts accepting updatedUser.event as a single parameter, which is how I ended up with "venue" stuff and VS Code stopped complaining. I couldn't figure out what was the data type of updatedUser.event. I should've done what I've always done and looked to the interface EventInvite as an example for that venue name. Now resolver.ts can stop complaining.

**Comment by ttran913** [comment]
I took your suggestions and hopefully made it more stylistically consistent with the rest of the file. Also removed all the unnecessary "venue as a whole data type" thing. Your "venue: { name: string | undefined } | null;" made resolver.ts stop complaining.

**Review by gikf — APPROVED** [review]
LGTM :+1:

**Comment by Sboonny** [comment]
#### :sparkles: :tada: **AWESOME!** :tada: :sparkles:

 Hi @ttran913, thanks for this pull request and for contributing to the code-base for the first time. We are looking forward to more contributions from you in the future.

 Cheers & happy contributing!

**Comment by Sboonny** [comment]
@all-contributors would you be so kind and add @ttran913 for code?

**Pull request merged**